the-collab-lab/tcl-30-whats-near-me
# jw-mm-update-app-details

## Description

As a user of a mobile device, I want to be able to install the app on my mobile device's home screen so that I can access it as if it’s a native app.



## Related Issue

Closes [TCL30-8](https://the-collab-lab.atlassian.net/browse/TCL30-8)


## Acceptance Criteria

Once this story is complete, users should be able to install the app on their device home screens, identifiable as our app, not the generic create-react-app app.

[Progressive Web App documentation](https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps)

The tags/data points that should be updated include the following:

The following tags/data points have been updated:

`public/index.html`

- `<meta name="theme-color">`

- `<meta name="description">`

- `<title>`

`public/manifest.json`

- `"short_name"`

- `"name"`

- `"icons"`

- `"theme_color"`

- `"background_color"`



## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

To create an icon, we used [Font Awesome](https://fontawesome.com/) Free Icons. We decided on `map-marked-alt-solid.svg` so we could have a transparent background.

To create the Apple Touch Icon (the name for the app icon on a mobile device), we learned that [Apple does not support `.SVG` for this purpose](https://stackoverflow.com/questions/15710047/apple-touch-icon-svg-support). Megan was able to quickly convert the icon into a `.PNG` using Adobe Illustrator. 

We also used [this medium article](https://medium.com/swlh/are-you-using-svg-favicons-yet-a-guide-for-modern-browsers-836a6aace3df) for reference. 

### Before
Screenshot on mobile (iPhone 8)     |  Icon on main screen on mobile (iPhone 8)    |  Screenshot on laptop including tab
:----------------------------------:|:--------------------------------------------:|:-------------------------------------:
<img width="300" alt="app_before" src="https://user-images.githubusercontent.com/38255956/125213310-8547e880-e277-11eb-9470-ae5ebc175f4b.PNG">  |  <img width="300" alt="update_before" src="https://user-images.githubusercontent.com/38255956/125213317-8b3dc980-e277-11eb-880a-2a018b63dce8.PNG"> | <img width="300" alt="before_computer_mobile_tab" src="https://user-images.githubusercontent.com/38255956/125214099-df967880-e27a-11eb-863c-e8c8708f392b.png">


### After
Screenshot on mobile (iPhone 8)     |  Icon on main screen on mobile (iPhone 8)    |  Screenshot on laptop including tab
:----------------------------------:|:--------------------------------------------:|:-------------------------------------:
<img width="300" alt="app_after" src="https://user-images.githubusercontent.com/38255956/125213936-1750f080-e27a-11eb-9bc9-41fd2319fc79.PNG">  |  <img width="300" alt="update_after" src="https://user-images.githubusercontent.com/38255956/125213941-1ddf6800-e27a-11eb-867e-395b82b84c5b.PNG"> | <img width="300" alt="after_computer_mobile_tab" src="https://user-images.githubusercontent.com/38255956/125214122-f1781b80-e27a-11eb-8025-4e32ba342f7c.png">



## Testing Steps / QA Criteria

1. With `whats-near-me` on your VS Code, navigate to your VS Code terminal
2. Run `git checkout jw-mm-update-app-details`
3. Run `npm start`
4. Go to `localhost:3000` to see new logo and content on the window, and new logo and content on the tab
5. To check on your mobile device you will need to do the following steps:
6. Keep the app running on your computer!
7. Make sure your mobile device and your computer are on the same wi-fi
8. On your computer, find your IP Address. The following directions will guide you if you are using a MacBook and Safari on an iOS device. 
9. On the computer running the app, go to `System Preferences`, `Network`, and you will see your IP Address listed

<img width="600" alt="ipaddress" src="https://user-images.githubusercontent.com/38255956/125214697-0bffc400-e27e-11eb-87e8-654f240728fd.png">

10. Now, on your mobile device, open a new browser window
11. In the address bar, enter your IP Address, followed by ":3000" (react's default port) or the port number your app is using.
- example: 192.168.0.27:3000
12. Go!
13. **For iOS, in Safari**
- Once the page loads, click the icon on the bottom of the screen that is a square with an arrow pointing up out of it.
- Scroll down and select `Add to Home Screen`
- `Add`

**For Android, in Chrome**

- tap the three dots at the top right of the window.
- in the pull down menu, select `Add to Home Screen`
- `Add`
14. Now you can see the icon for "What's Near Me" alongside your other app icons!
